simtools/layout/layout_array.py

```diff
@@ -778,10 +844,33 @@ def get_telescope_type(telescope_name):
 
         _class, _ = names.split_telescope_model_name(telescope_name)
         try:
-            if _class[0:3] in ("LST", "MST", "SST", "SCT"):
+            if _class[0:3] in (names.all_telescope_class_names):
                 return _class[0:3]
 
         except IndexError:
             pass
 
-        return ""
+    @staticmethod
+    def include_radius_into_telescope_table(telescope_table):
```

Comment:
This function is not used anywhere but in tests and also seems strange to me. I suggest to remove it, but we can discuss it.

Reply:
same as above, I believe we will need this to the plotting application

Reply:
Technically for plotting you don't need to add a column to the table, you can simply read the radius of each telescope from the metadata in the table (or the DB). If you really want to make the plotting function not dependent on the metadata or the DB and want to keep this function, I won't object. However, in that case, please don't use `tel_name_now[:3]` to get the telescope class but the actual function you added. Also, I really don't understand why you need to use `.split()[0]` and `.split()[1]` to get the radius from the metadata. I thought the metadata is a dictionary.

Reply:
Indeed I prefer to keep it. I will change the tel_name_now as you suggest

the split is because the telescope name comes as MST-10, then we want only the first part. the second split is to get only the string for the unit to be passed to u.Unit. I dont think astropy would understand if we give directly the string such as `u.Quantity("10 m")` or would it?

Reply:
Wait, but if I understand the code correctly, in this function you refer to the metadata dictionary, not the list of telescopes. So in this line below
```
telescope_table.meta["corsika_sphere_radius"][tel_name_now[:3]].split()[0]
``` 
should actually be 
```
telescope_table.meta["corsika_sphere_radius"][get_telescope_class(tel_name_now)]["value"]
```
and for the unit it should be the same but with unit
```
telescope_table.meta["corsika_sphere_radius"][get_telescope_class(tel_name_now)]["unit"]
```
(right now you are always using the LST unit which should be fine, but not the best practice.

Reply:
This does not work. I think you are mixing up with the corsika_parameters.yml where we do have "value" and "unit".
Here, if I print the following for instance:
```
        print(type(telescope_table.meta["corsika_sphere_radius"]))
        print((telescope_table.meta["corsika_sphere_radius"]))
```
I get the following:
```
<class 'dict'>
{'LST': '12.5 m', 'MST': '9.6 m', 'SST': '3 m'}
```
Therefore, there is no "value" and "unit" separated in the metadata. We have to get them manually.

```

Reply:
Regarding the unit, I agree that it is not the best way. Nevertheless, we cannot pass all the units since we need the unit of the column "radius" in the table, regardless of the type of telescope. In this case, I changed the code to check if the units given in the metadata are the same for the telescope types and extracted one of them to set the table column unit. Please have a look, if you like the way it is I will implement the test for this new raise statement.

Reply:
OK, this issue made us realise that the layout files were not all the same. The metadata in one of the files was different to the others. This is now fixed and I think my code suggestion should therefore work now.